rust/ql/lib/codeql/rust/internal/PathResolution.qll

```diff
@@ -595,10 +595,10 @@ class ImplItemNode extends ImplOrTraitItemNode instanceof Impl {
     exists(TraitItemNode trait |
       trait = this.resolveTraitTy() and
       trait.hasCanonicalPath(c2) and
-      if this.hasCanonicalPath(c2)
-      then c1 = c2
-      else (
-        c2 = c1.getADependency() or c1 = c2.getADependency()
+      (
+        c1 = c2 or
+        c2 = c1.getADependency() or
+        c1 = c2.getADependency()
       )
```

Comment:
Parens can be removed if you replace the preceding `and` with `|`.

Reply:
I think I found a fix that matches better with the original intent of the code.